lib/nuget-parser/dotnet-core-parser.js

```diff
@@ -123,38 +121,41 @@ function getTargetObjToRun(manifest) {
   return manifest.targets[selectedTargetKey];
 }
 
+module.exports = {
+  parse: function (tree, fileContent, fileContentParser) {
+    return new Promise(function parseFileContents(resolve) {
+      debug('Trying to parse dot-net-cli manifest');
+      const manifest = fileContentParser.parse(fileContent);
 
-function parse(fileContent, packageTree) {
-  const manifest = JSON.parse(fileContent);
-
-  if (!manifest.project) {
-    throw new Error('Project field was not found in project.assets.json');
-  }
-
-  if (manifest.project.version) {
-    packageTree.version = manifest.project.version;
-  }
+      if (!manifest.project) {
+        throw new Error('Project field was not found in project.assets.json');
```

Comment:
`return reject(new Error('Project field was not found in project.assets.json'));`

Reply:
It should work both ways, but this looks more natural in case of promises.

Reply:
Changing